src/core/control/tools/ImageHandler.cpp

```diff
@@ -84,15 +104,13 @@ auto ImageHandler::insertImage(GFile* file, double x, double y) -> bool {
 
     img->setWidth(width * zoom);
     img->setHeight(height * zoom);
+}
 
-    page->getSelectedLayer()->addElement(img);
-
-    control->getUndoRedoHandler()->addUndoAction(
-            std::make_unique<InsertUndoAction>(page, page->getSelectedLayer(), img));
-
-    view->rerenderElement(img);
-    auto* selection = new EditSelection(control->getUndoRedoHandler(), img, view, page);
-    control->getWindow()->getXournal()->setSelection(selection);
-
-    return true;
+auto ImageHandler::insertImage(GFile* file, double x, double y) -> bool {
+    auto [img, width, height] = ImageHandler::createImage(file, x, y);
+    if (!img) {
+        return false;
+    }
+    ImageHandler::automaticScaling(img, x, y, width, height);
+    return ImageHandler::addImageToDocument(img, true);
```

Comment:
I don't see the need for scoping here.
```suggestion
    automaticScaling(img, x, y, width, height);
    return addImageToDocument(img, true);
```